### ISSUE TITLE
Add version  & CDN info to plugin documentation

### ISSUE THREAD

**Issue by jodeleeuw** [jspsych/jsPsych#2521]
It's not clear on the plugin pages what the current version of the plugin is that should be loaded from the CDN. I think we should add an Install section to each plugin page with a `<script>` tag for the current version and the `npm install` command.

**Comment by bjoluc** [comment]
Absolutely! The unpkg URL will automatically be updated to the latest version then on releases :tada:

**Comment by becky-gilbert** [comment]
@jodeleeuw I created a branch for this called `plugin-docs-add-version` and added a basic Install section to the [animation doc page](https://github.com/jspsych/jsPsych/blob/plugin-docs-add-version/docs/plugins/animation.md). I made a quick draft with the Install section in between Data Generated and Examples. Here's how it looks in the TOC at the top of the page:

![animation-install-1](https://user-images.githubusercontent.com/9041788/165619633-89c9835b-990f-429d-95cd-2c66114c683f.PNG)
And here's what the Install section looks like:

![animation-install-2](https://user-images.githubusercontent.com/9041788/165623481-9f9eed40-6c40-4c4d-b4ad-10437f60a34d.PNG)

Is this what you had in mind? And are the file paths and NPM lines correct?

Also, is it ok if I add a gulp task to get the latest version number and pipe it in somewhere on each package docs page? I'm thinking I can use the [`updateUnpkgLinks` function](https://github.com/jspsych/jsPsych/blob/5cf4e5bc093749b0dd8601f9fdc5373447dca3b4/packages/config/gulp.js#L111-L129) as a reference, and do something similar to replace all instances of e.g. "\<package-name\>-current-version" with the current version number in the markdown files. I thought I could either add it to the page title `<h1>` after the plugin name, e.g. "animation v1.1.0", or add something like "Current version: 1.1.0" to the top of the Install section. Thoughts?

**Comment by jodeleeuw** [comment]
This looks great! 

What if we also add a link to the changelog, e.g., here's the log for the virtual chinrest plugin that I recently updated: https://github.com/jspsych/jsPsych/blob/main/packages/plugin-virtual-chinrest/CHANGELOG.md.

Maybe we do something like

---

### animation

Current version: 1.1.0. [See version history.](https://github.com/jspsych/jsPsych/blob/main/packages/plugin-animation/CHANGELOG.md)

This plugin displays a sequence of images at a fixed frame rate. ...

---

Or we could put this at the top of the install section. I'm not sure which is smoother.

Adding a gulp task to update this sounds perfect.

**Comment by becky-gilbert** [comment]
Yes! I'll add the version history. I think the top of the page looks good. I'll do this for all plugin docs pages. It'll be easy to move things around later if you want.

Great, I'll also try adding a gulp task to automatically update the version numbers.

**Comment by bjoluc** [comment]
A use case for badges (e.g. via https://shields.io/)?
`![current version](https://img.shields.io/npm/v/@jspsych/plugin-animation?label=Current%20version)`
makes
![current version](https://img.shields.io/npm/v/@jspsych/plugin-animation?label=Current%20version)
and gets the latest version from NPM.
Pro: Old minor docs =(jsPsych) versions – although not updated anymore – will still show the up-to-date version, which is typically compatible.
Con: So do the 7.x docs when the plugins support jsPsych v8 only...

Interestingly, shields can also do
`![required jsPsych version](https://img.shields.io/npm/dependency-version/@jspsych/plugin-animation/peer/jspsych?label=Required%20jsPsych%20version)`
becoming
![required jsPsych version](https://img.shields.io/npm/dependency-version/@jspsych/plugin-animation/peer/jspsych?label=Required%20jsPsych%20version)

Just dropping this here as an idea; maybe static versions are the better option.

**Comment by becky-gilbert** [comment]
Oooh very cool, thanks @bjoluc! What do you think @jodeleeuw?

I guess I hadn't thought about what happens with the plugin current versions in the versioned docs once a new jsPsych release comes out... I can see the problem here. I guess one solution would be to manually change the badge into a static version number anytime there's a break in compatibility. For example, if the latest plugin version is v2.0.0 but it requires jsPsych v7.2+, then in the jsPsych v7.1 docs, instead of the badge it would say something like "Last compatible version: v1.1.0". Or we could at least add the "required jsPsych version" badge to the page so that it would indicate to users that the current plugin version isn't compatible with the jsPsych version that they're using, but then it'd be up to the user to notice this and then figure out what plugin version they need to use...

**Comment by becky-gilbert** [comment]
I just updated the `plugin-docs-add-version` branch. It now has the following on every `docs/plugins/*.md` page:

- current version number
- link to version history (changelog) 
- Install section

And I added a gulp task to auto-add/update the current plugin versions in the docs, using the version numbers in the `package.json` files (i.e. same way that the unpkg links are updated, including those in the new Install sections). I realize we might switch to using badges but I was pretty much finished with this anyway 😃 

I ran the gulp `updatePluginVersion` task before committing, so all of the plugin version numbers in the `docs/plugins/*.md` pages should be correct, assuming my branch is still update with `main`. I also tested this by changing the version number in one of the plugin's `package.json` file and re-running the gulp tasks.
